app/src/main/java/net/gsantner/markor/model/Document.java

```diff
@@ -100,7 +104,32 @@ public File getFile() {
         return _file;
     }
 
-    public long lastModified() {
+    private void initModTimePref() {
+        // We do not do this in constructor as we want to init after deserialization too
+        if (_modTimePref == null) {
+            _modTimePref = App.get().getApplicationContext().getSharedPreferences(MOD_PREF_NAME, Context.MODE_PRIVATE);
+        }
+    }
+
+    private long getGlobalTouchTime() {
+        initModTimePref();
+        return _modTimePref.getLong(_file.getAbsolutePath(), 0);
+    }
+
+    private void setGlobalTouchTime() {
+        initModTimePref();
+        _touchTime = System.currentTimeMillis();
+        _modTimePref.edit().putLong(_file.getAbsolutePath(), _touchTime).apply();
+    }
+
+    public boolean hasFileChangedSinceLastLoad() {
+        return _modTime < 0                            // Never read
+                || _touchTime < 0                      // Never read
+                || fileModTime() > _modTime            // File mod time updated
+                || getGlobalTouchTime() > _touchTime;  // File has been modified by another document
```

Comment:
If we hit this line and the result is _true_ then the mod time is in a bad state. We should log it

Reply:
Fine to add a few `Log.`'s at the central important places